core/parse-xml.c

```diff
@@ -558,6 +558,13 @@ static void hex_value(char *buffer, uint32_t *i)
 	*i = strtoul(buffer, NULL, 16);
 }
 
+static void dive_site(char *buffer, struct dive_site **ds)
+{
+	uint32_t uuid;
+	hex_value(buffer, &uuid);
+	*ds = get_dive_site_by_uuid(uuid);
```

Comment:
this dereference doesn't need a NULL check?

Reply:
None of the output parameters in the parser are checked for NULL. Therefore I would think that for the sake of consistency we shouldn't add a NULL-check here either.